freqtrade/rpc/api_server/api_ws.py

```diff
@@ -0,0 +1,61 @@
+import logging
+
+from fastapi import APIRouter, Depends, WebSocket, WebSocketDisconnect
+
+from freqtrade.enums import RPCMessageType
+from freqtrade.rpc.api_server.deps import get_channel_manager
+from freqtrade.rpc.api_server.ws.utils import is_websocket_alive
+
+
+logger = logging.getLogger(__name__)
+
+# Private router, protected by API Key authentication
+router = APIRouter()
+
+
+@router.websocket("/message/ws")
+async def message_endpoint(
+    ws: WebSocket,
+    channel_manager=Depends(get_channel_manager)
+):
+    try:
+        if is_websocket_alive(ws):
+            logger.info(f"Consumer connected - {ws.client}")
+
+            # TODO:
+            # Return a channel ID, pass that instead of ws to the rest of the methods
+            channel = await channel_manager.on_connect(ws)
+
+            # Keep connection open until explicitly closed, and sleep
+            try:
+                while not channel.is_closed():
+                    request = await channel.recv()
+
+                    # This is where we'd parse the request. For now this should only
+                    # be a list of topics to subscribe too. List[str]
+                    # Maybe allow the consumer to update the topics subscribed
+                    # during runtime?
+
+                    # If the request isn't a list then skip it
+                    if not isinstance(request, list):
+                        continue
+
+                    # Check if all topics listed are an RPCMessageType
+                    if all([any(x.value == topic for x in RPCMessageType) for topic in request]):
+                        logger.debug(f"{ws.client} subscribed to topics: {request}")
+                        channel.set_subscriptions(request)
```

Comment:
I think to future-proof ourselfs, we'd want a slightly different approach here.
Right now, we assume "if what is sent is a list - it has to be a new list of subscriptions"
i think this will limit future usability - this should be a (small) protocol similar to the RPCMessageType object - which defines a type - and a message
something around 

```
{
   'type': 'subscribe',
   'data': ['whitelist', 'greenlist', ...]
}
```

that way you can have diff. message types - where the client can subscribe to different topics (whitelist, entries, ...) - or request info (similar to sending `/daily`) - although i'd not implement this in detail for now (it just shouldn't be prevented by the protocol definition early on)

Reply:
So the reasoning behind this was if for whatever reason a malicious party had access to the `ws_token` and was sending bad messages to the websocket that hopefully this would be sufficient to ignore. I agree with having a `RequestMessage` of sorts that the consumer can send to the producer, however actually implementing the requests other than the subscription topics should not happen until later on.

Reply:
That's correct - i don't see implementation of this for this PR. 

But using a list now will prevent us in the future - and that's my point here :) 

---

Regarding an attacker ... we should clearly not call "eval" (or smiilar) on info sent - but simple parsing of "recv.get('msgtype') == Whatever.xxx" should not open up an attack surface.

Reply:
Check rpc.py > `RPC._process_consumer_request`

Reply:
great, that's exactly what i mean.

If i'm honest, i'd probably keep this in the WS file (where it's called).
it's specific code that's only used by websockets (there won't be another rpc handler that has exactly the same requirements).
that's nitpicks though - as it won't impact functionality - so i'm perfectly fine with leaving it where it is for now.

Reply:
:+1: